### PR TITLE
Chore/540 migrate terraform

### PR DESCRIPTION
**Pull request by joshgamache**
Addresses #540 - [Project link](https://github.com/orgs/bcgov/projects/123/views/2?pane=issue&itemId=51794032)
Adds Terraform scripts and Helm charts to run them. Resource states migrated from Terraform Cloud to a Google Cloud Storage bucket.

## Changes 🚧

- Added Terraform module, imported from cas-shelf
- Added Helm templates to handle:
  - Terraform module loading to configMaps
  - OpenShift service account for use by Terraform
  - Terraform apply job

## To test 🔬

- The Helm charts are built into the automation around deploys via ShipIt. Once a deploy has been run, you can check the job `Terraform-apply` in OpenShift to see if the run was successful.

## Notes📝 
- 🔒 Terraform Cloud state has been locked to enable migration of state to state buckets
- A script has been run and resources have been transitioned from Terraform Cloud to GCP buckets holding state for each namespace (ggircs-dev, ggircs-test, ggircs-prod).  
- Resources have been migrated rather than recreated, only the state "owning" the resource changes. 
- The resources `kubernetes_secret.secret_sa` have had Terraform changes, but only the addition of a default value (consistent with our use of them) due to a new provider version. 
- Unlike other project's migrations, the one set of buckets created based on 'apps' are _different_ between `dev` and `test` + `prod`, due to name length limits. This app is `eccc-swrs-import` in dev and `swrs-import` in test+prod.

### CODE REVIEW

File: helm/cas-ggircs/.helmignore [hunk 1]
```diff
@@ -21,3 +21,11 @@
 .idea/
 *.tmproj
 .vscode/
+
```
Comment (mikevespi) [inline helm/cas-ggircs/.helmignore:24]: Let's add these to the gitignore too

File: helm/cas-ggircs/values.yaml [hunk 2]
```diff
@@ -163,3 +163,20 @@ nginx-sidecar:
...
+  resources:
+    limits:
+      cpu: 1000m
+      memory: 512Mi
+    requests:
+      cpu: 100m
+      memory: 64Mi
+
+  terraform:
+   namespace_apps: '["ggircs-backups", "swrs-import", "mb-backups"]'
```
Comment (mikevespi) [inline helm/cas-ggircs/values.yaml:182]: App names in `values-dev` aren't quite in line, is this intentional?

**Review by mikevespi — DISMISSED** [review]
Aside from the couple of comments, good to go! 🎉 
Leaving unmerged per todays discussion.

**Review by JoshLarouche — APPROVED** [review]

File: helm/cas-ggircs/values-dev.yaml [hunk 3]
```diff
@@ -32,3 +32,7 @@ persistence:
 
 nginx-sidecar:
   hostName: cas-ggircs-dev.apps.silver.devops.gov.bc.ca
+
+devops:
+  terraform:
+   namespace_apps: '["ggircs-backups", "eccc-swrs-import", "mb-backups"]'
```
Comment (pbastia) [inline helm/cas-ggircs/values-dev.yaml:38]: `mb-backups` is for metabase, which is its own chart/repository: https://github.com/bcgov/cas-metabase
It also configures a separate instance of `cas-postgres`, with its own backup buckets: https://github.com/bcgov/cas-metabase/blob/develop/helm/cas-metabase/values-dev.yaml

```suggestion
   namespace_apps: '["ggircs-backups", "eccc-swrs-import"]'
```

**Review by pbastia — CHANGES_REQUESTED** [review]
We need to take the metabase bucket out of this TF run and add it to the metabase chart instead

File: helm/cas-ggircs/values-dev.yaml (hunk 3 above)
Comment (joshgamache) [reply helm/cas-ggircs/values-dev.yaml:38]: I'll go in and re-import the state so it doesn't include _`mb-backups`_. Then we'll need to do a migration on `cas-metabase` for it to own it's own state.

**Comment by pbastia** [comment]
This one should also be refactored to use the chart instead, there's no real point in merging this work anymore

**Review by pbastia — APPROVED** [review]
👌 👌

**Pull request merged**